### DISCUSSION TITLE
EVCC - Charge your Tesla with Solar

### DISCUSSION THREAD

**Discussion by Briochorama** [tesla-local-control/tesla_ble_mqtt_docker#12] [Show and tell]
>[!NOTE]
> This is a discussion about an external addon.

In a separate discussion about the HA integration for Tesla, [Raphmur mentionned stumbling upon EVCC](https://github.com/alandtse/tesla/issues/961#issuecomment-2169690117), a tool that allows users to charge their vehicle using solar power (and much more apparently).

I'm currently using a DIY solution for this, and also it works, it looks nothing like what their team seems to have achieved. I thought this would be interesting for some users here. Once I've read the full documentation, I'll update this post with a config that can your with the BLE commands available from this repo.

**Comment by raphmur** [comment]
Hello @MrBLJ, I also considered making my variables and solar regulation into an integration (which uses the ble docker version or the HA addon version). Did you progress on this?

**Reply by Briochorama** [reply]
Hi,

I did progress indeed ! I've been using EVCC for the past months now, and it's been working flawlessly using tesla BLE commands..

I'm not sure I understand you question though, are you looking for an EVCC config to use with the HA addon, or something else ?
